### DISCUSSION TITLE
formily1.x版本中，submit时获取的表单值可以是object嵌套的吗？

### DISCUSSION THREAD

**Discussion by zhh877644371** [alibaba/formily#1284] [Q&A 1.x] [answered]
利用布局组件实现的多层嵌套的schema解构，在每一个层级定义了key，在获取表单值的时候可以按照schema的结构，获取多层object嵌套的value吗？
例如schema是这种结构：
`{
  "type": "object",
  "properties": {
    "key1": {
      "type": "object",
      "properties": {
        "key2": {"type": "string"}
       }
     }
  }
}`
可以获取到如下的值结构吗：
`
{key1: {key2: xxx}}
`

**Comment by janryWang** [comment, marked as answer]
可以的，为啥不用2.0

**Reply by zhh877644371** [reply]
2.0有一定的改造成本，暂时排期有限，如果1.x没办法实现的话，可能会考虑升级2.0

**Reply by zhh877644371** [reply]
怎么实现呢，没在文档里看到

**Reply by janryWang** [reply]
就像你上面给出的schema，传给SchemaForm就行了

**Reply by zhh877644371** [reply]
我现在就是这样做的，但是我看到value的值平铺的，而且子节点的key也不能重复

**Reply by janryWang** [reply]
写个codesandbox看看吧

**Reply by zhh877644371** [reply]
测试了一下，发现原来是我用了block组件，会忽略那一层的key，有什么好的处理方法吗？我现在想到的是只能多包一层空的type:'object'结构

**Reply by janryWang** [reply]
多包一层是标准方案
